app/src/main/java/clarkson/ee408/tictactoev4/PairingActivity.java

```diff
@@ -68,58 +86,166 @@ protected void onCreate(Bundle savedInstanceState) {
+
+                AppExecutors.getInstance().mainThread().execute(() ->
+                        handlePairingUpdate(pr)
+                );
+
+            } catch (Exception e) {
+                Log.e(TAG, "Error updating pairing", e);
+            }
+        });
     }
 
     /**
      * Handle the PairingResponse received from the server
+     *
      * @param response PairingResponse from the server
      */
     private void handlePairingUpdate(PairingResponse response) {
-        // TODO: handle availableUsers by calling updateAvailableUsers()
 
-        // TODO: handle invitationResponse. First by sending acknowledgement calling sendAcknowledgement()
-        // --TODO: If the invitationResponse is ACCEPTED, Toast an accept message and call beginGame
-        // --TODO: If the invitationResponse is DECLINED, Toast a decline message
+        //Handle availableUsers by calling updateAvailableUsers()
+        updateAvailableUsers(response.getAvailableUsers());
+
+        //Handle invitationResponse. First by sending acknowledgement
+        Event invitationResponse = response.getInvitationResponse();
+        if (invitationResponse != null) {
+
+            // acknowledge it
+            sendAcknowledgement(invitationResponse);
 
-        // TODO: handle invitation by calling createRespondAlertDialog()
+            //If ACCEPTED → Toast + beginGame()
+            if (invitationResponse.getStatus() == EventStatus.ACCEPTED) {
+                Toast.makeText(this, invitationResponse.getSender() + " accepted your request!", Toast.LENGTH_SHORT).show();
+                beginGame(invitationResponse, 1);
+
+            } else if (invitationResponse.getStatus() == EventStatus.DECLINED) {
+                //If DECLINED → Toast message
+                Toast.makeText(this, invitationResponse.getSender() + " declined your request.", Toast.LENGTH_SHORT).show();
+            }
+        }
+
+        //Handle invitation by calling createRespondAlertDialog()
+        if (response.getInvitation() != null) {
+            createRespondAlertDialog(response.getInvitation());
+        }
     }
 
     /**
      * Updates the list of available users
+     *
      * @param availableUsers list of users that are available for pairing
      */
     public void updateAvailableUsers(List<User> availableUsers) {
         adapter.setUsers(availableUsers);
+
         if (adapter.getItemCount() <= 0) {
-            // TODO show noAvailableUsersText and hide recyclerView
+            //Show noAvailableUsersText and hide recyclerView
+            noAvailableUsersText.setVisibility(View.VISIBLE);
+            recyclerView.setVisibility(View.GONE);
         } else {
-            // TODO hide noAvailableUsersText and show recyclerView
+            //Hide noAvailableUsersText and show recyclerView
+            noAvailableUsersText.setVisibility(View.GONE);
+            recyclerView.setVisibility(View.VISIBLE);
         }
     }
 
     /**
      * Sends game invitation to an
```

Comment:
Finish the incomplete javadoc from the original code

```suggestion
     * Sends game invitation to an opponent.
```